core/domain/suggestion_services_test.py

```diff
@@ -1579,6 +1579,50 @@ def test_remove_exp_from_story_rejects_translation_suggestion(self):
         self.assertEqual(
             suggestions[0].status, suggestion_models.STATUS_REJECTED)
 
+    def test_translation_update(self):
+        self.create_translation_suggestion_associated_with_exp(
+            self.EXP_ID, self.author_id)
+        suggestions = suggestion_services.query_suggestions(
+            [('author_id', self.author_id), ('target_id', self.EXP_ID)])
+
+        suggestion_services.update_suggestion(
+            suggestions[0].suggestion_id, '<p>Test Translation</p>')
+
+        updated_suggestion = suggestion_services.get_suggestion_by_id(
+            suggestions[0].suggestion_id)
+        self.assertEqual(
+            updated_suggestion.change.translation_html,
+            '<p>Test Translation</p>')
+
+    def test_translation_update_not_existing_exception(self):
+        self.create_translation_suggestion_associated_with_exp(
+            self.EXP_ID, self.author_id)
+        suggestions = suggestion_services.query_suggestions(
+            [('author_id', self.author_id), ('target_id', self.EXP_ID)])
+        suggestion_services.accept_suggestion(
+            suggestions[0].suggestion_id, self.reviewer_id, 'Done', 'Accepted'
+        )
+        expected_exception_regexp = (
+            'The suggestion with id %s has already been accepted/'
+            'rejected.' % (suggestions[0].suggestion_id)
+        )
+
+        with self.assertRaisesRegexp(
+            utils.ValidationError, expected_exception_regexp):
+            suggestion_services.update_suggestion(
+                suggestions[0].suggestion_id, '<p>Test Translation</p>')
+
+    def test_translation_update_already_hadled_exception(self):
+        expected_exception_regexp = (
+            'You cannot change the suggestion with id %s because it does not '
+            'exist.' % ('sug-1')
+        )
+
+        with self.assertRaisesRegexp(
+            utils.ValidationError, expected_exception_regexp):
+            suggestion_services.update_suggestion(
+                'sug-1', '<p>Test Translation</p>')
```

Comment:
Since these tests involve testing just the suggestions, let's move them to `SuggestionServicesUnitTests`.

Reply:
Done